lodash_exercise/lodash.js

```diff
@@ -1,103 +1,248 @@
-function drop(){
-
-}
-
-function fromPairs(){
-
-}
-
-function head(){
-
-}
-
-function take(){
-
-}
-
-function takeRight(){
-
-}
-
-function takeWhile(){
-
-}
-
-function union(){
-
+function drop(arr, num) {
+    if (num === undefined) { // or arguments[1] <-> num
+        return arr.slice(1);
+    }
+    num = num || 0;
+    return arr.slice(num);
+}
+
+function fromPairs(arr) {
+    var obj = {};
+    arr.forEach(function (el) {
+        obj[el[0]] = el[1];
+    });
+    return obj;
+}
+
+function head(arr) {
+    if (arr === []) return undefined;
+    return arr[0];
+}
+
+function take(arr, num) {
+    if (num === undefined) {
+        return arr.slice(0, 1);
+    }
+    return arr.slice(0, num);
+}
+
+function takeRight(arr, num) {
+    if (num === undefined) {
+        return arr.slice(-1);
+    }
+    if (num === 0) return [];
+    return arr.slice(-num);
+}
+
+function union(arr1, arr2) {
```

Comment:
how could you solve this problem if you have more than two parameters (3 arrays / 5 arrays etc.)?